primer/brand
# Hide paged items in Pagination component on narrow viewports

## Summary

Resolves https://github.com/github/primer/issues/3744

Numbered pagination items are now hidden on narrow viewports by default. This prevents horizontal overflow / scrolling issues when a large number of items are present. 

While the API already supports `showPages`, this enforces that behavior by default while retaining the option to customize it.

🔗  [Storybook preview](https://primer-ce1b22441a-26139705.drafts.github.io/brand/storybook/?path=/story/components-pagination-features--narrow-page-numbers-hidden-by-default&globals=viewport:iphonexr)

## List of notable changes:



- `showPages` now undefined by default (prev `true`)
- `showPages` is `true` if viewport is > `isMedium`
- `showPages` continues to be customizable and forwarded by end-users (backwards compatible)

## What should reviewers focus on?



- Are there any edge cases that haven't been handled?

## Steps to test:



1. [Use the dedicated story](https://primer-ce1b22441a-26139705.drafts.github.io/brand/storybook/?path=/story/components-pagination-features--narrow-page-numbers-hidden-by-default&globals=viewport:iphonexr) to see new default behaviour
1. Scroll through all other stories and check the behavior matches the expectations (hidden on mobile, still visible on medium+)

## Supporting resources (related issues, external links, etc):

- https://github.com/github/primer/issues/3744

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

![image](https://github.com/user-attachments/assets/518c31da-3a71-4878-b85c-33bb2b50437d)


 </td>
<td valign="top">

![Screenshot 2024-08-23 at 11 05 31](https://github.com/user-attachments/assets/b19c4490-3ced-41e9-8ab0-2f01511b8e13)


</td>
</tr>
</table>
